### ISSUE TITLE
Sensor: Allow exit conditions when a sync is not running, will not run for some duration of time

### ISSUE THREAD

**Issue by crw** [fivetran/airflow-provider-fivetran#37]
Currently the sensor will exit successfully after "the next" sync job completes successfully. The basic assumption is that there is a sync job already running.

We need to add parameters to the sensor that in effect allow the user to specify:

"If the sync ran within x amount of time AND
the sync is not scheduled to start within y amount of time THEN
exit successfully."

**Comment by scoyne2** [comment]
@crw do you think this is the same thing? I am running a Fivetran Operator to Trigger a sync, then a Fivetran Sensor to wait on the sync to complete. The Sensor correctly continues to poke when the Fivetran status is "syncing', but when the sync completes and the status changes to `scheduled` the sensor still continues to sync (I would expect it to mark the FivetranSensor task as success


```
[2021-05-01 21:35:16,319] {taskinstance.py:867} INFO - Starting attempt 5 of 6
[2021-05-01 21:35:16,319] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2021-05-01 21:35:16,343] {taskinstance.py:887} INFO - Executing <Task(FivetranSensor): fivetran_sensor> on 2021-04-29T13:00:00+00:00
[2021-05-01 21:35:16,352] {standard_task_runner.py:53} INFO - Started process 32825 to run task
[2021-05-01 21:35:16,497] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: salesforce.fivetran_sensor 2021-04-29T13:00:00+00:00 [running]> 2c399c08984c
[2021-05-01 21:35:19,617] {logging_mixin.py:112} INFO - [2021-05-01 21:35:19,617] {fivetran.py:282} INFO - Connector "wording_activity": sync_state = syncing
[2021-05-01 21:35:19,679] {taskinstance.py:1079} INFO - Rescheduling task, marking task as UP_FOR_RESCHEDULE
[2021-05-01 21:35:26,202] {logging_mixin.py:112} INFO - [2021-05-01 21:35:26,201] {local_task_job.py:103} INFO - Task exited with return code 0
[2021-05-01 21:41:19,031] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: salesforce.fivetran_sensor 2021-04-29T13:00:00+00:00 [queued]>
[2021-05-01 21:41:19,104] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: salesforce.fivetran_sensor 2021-04-29T13:00:00+00:00 [queued]>
[2021-05-01 21:41:19,105] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2021-05-01 21:41:19,105] {taskinstance.py:867} INFO - Starting attempt 5 of 6
[2021-05-01 21:41:19,105] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2021-05-01 21:41:19,132] {taskinstance.py:887} INFO - Executing <Task(FivetranSensor): fivetran_sensor> on 2021-04-29T13:00:00+00:00
[2021-05-01 21:41:19,140] {standard_task_runner.py:53} INFO - Started process 36687 to run task
[2021-05-01 21:41:19,348] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: salesforce.fivetran_sensor 2021-04-29T13:00:00+00:00 [running]> 2c399c08984c
[2021-05-01 21:41:23,167] {logging_mixin.py:112} INFO - [2021-05-01 21:41:23,167] {fivetran.py:282} INFO - Connector "wording_activity": sync_state = scheduled
[2021-05-01 21:41:23,227] {taskinstance.py:1079} INFO - Rescheduling task, marking task as UP_FOR_RESCHEDULE
[2021-05-01 21:41:28,851] {logging_mixin.py:112} INFO - [2021-05-01 21:41:28,850] {local_task_job.py:103} INFO - Task exited with return code 0
```

**Comment by crw** [comment]
@scoyne2 sorry for the late reply. This was a bug in versions earlier that 1.0.1, can you try upgrading to the latest version and see if you still have this problem?

**Comment by scoyne2** [comment]
Hi @crw I am on v1.0.1 seeing this issue. Thanks!

**Comment by scoyne2** [comment]
I believe it is an issue with mode="reschedule", I will try mode="poke" and see if that resolves the issue.

**Comment by crw** [comment]
Ah yes, this is a known issue. We should file a new issue for this problem. We may get a patch soon, but in any case in reschedule mode, we need to let the user have the ability to pass a value for the time of the previously completed sync -- in other words, any sync that finishes after the specified time would be considered to be a completed sync. Basically, the Sensor object is storing a little state and if the sensor is destroyed, we lose that state and never correctly exit. Thanks for the additional details there!

**Comment by RoshanAnchan** [comment]
@crw , when the patch or new version of the package will release?

**Comment by raajpackt** [comment]
Cannot i just mark the sensor as successful when sync is not running?
